java/cuvs-java/src/test/java/com/nvidia/cuvs/CagraRandomizedIT.java

```diff
@@ -85,23 +85,37 @@ private void tmpResultsTopKWithRandomValues() throws Throwable {
       }
     }
     // Sanity checks
-    assert dataset.length > 0 : "Dataset is empty.";
+    assert vectors.length > 0 : "Dataset is empty.";
     assert queries.length > 0 : "Queries are empty.";
     assert dimensions > 0 : "Invalid dimensions.";
     assert topK > 0 && topK <= datasetSize : "Invalid topK value.";
 
     // Generate expected results using brute force
-    List<List<Integer>> expected = generateExpectedResults(topK, dataset, queries, null, log);
+    List<List<Integer>> expected = generateExpectedResults(topK, vectors, queries, null, log);
 
     // Create CuVS index and query
     try (CuVSResources resources = CuVSResources.create()) {
       CagraIndexParams indexParams = new CagraIndexParams.Builder()
           .withCagraGraphBuildAlgo(CagraGraphBuildAlgo.NN_DESCENT)
           .build();
-      CagraIndex index = CagraIndex.newBuilder(resources)
-          .withDataset(dataset)
-          .withIndexParams(indexParams)
-          .build();
+
+      CagraIndex index;
+      boolean useNativeMemoryDataset = random.nextBoolean();
```

Comment:
Likewise here.  I wonder if we should test with both Vectors and with Datasets, one after the other.

Are we doing this to reduce test time?

Reply:
Yes, wanted to reduce the test time by making it randomized. But, it feels important enough to test twice.